### DISCUSSION TITLE
Installation issues

### DISCUSSION THREAD

**Discussion by lackovic** [aquaproj/aqua#2450] [Deprecated (Question and Answer)] [answered]
### Question

I installed `aqua` on Linux using the shell script taken from [the docs](https://aquaproj.github.io/docs/products/aqua-installer/):

```sh
$ curl -sSfL https://raw.githubusercontent.com/aquaproj/aqua-installer/v2.1.2/aqua-installer | bash
===> Installing aqua v2.9.0 for bootstrapping...
===> Downloading https://github.com/aquaproj/aqua/releases/download/v2.9.0/aqua_linux_amd64.tar.gz ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 6578k  100 6578k    0     0  7158k      0 --:--:-- --:--:-- --:--:-- 20.9M
===> Verifying checksum of aqua v2.9.0 ...
aqua_linux_amd64.tar.gz: OK
===> /tmp/tmp.rpZdKwO38b/aqua update-aqua
INFO[0000] download and unarchive the package            aqua_version=2.9.0 env=linux/amd64 new_version=v2.16.4 package_name=aquaproj/aqua package_version=v2.16.4 program=aqua registry=
INFO[0000] verify a package with slsa-verifier           aqua_version=2.9.0 env=linux/amd64 new_version=v2.16.4 package_name=aquaproj/aqua package_version=v2.16.4 program=aqua registry=
INFO[0000] download and unarchive the package            aqua_version=2.9.0 env=linux/amd64 new_version=v2.16.4 package_name=slsa-framework/slsa-verifier package_version=v2.3.0 program=aqua registry=
Verified signature against tlog entry index 47217472 at URL: https://rekor.sigstore.dev/api/v1/log/entries/24296fb24b8ad77ab372bccde8f255a5fecd26304fac30eee9c5e785f0cad4c383c2beeeacc563e2
Verified build using builder https://github.comslsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@refs/tags/v1.9.0 at commit fd3d2911afd67834ecdcf4c282fa294727cee9eb
Verifying artifact /tmp/760606590: PASSED

PASSED: Verified SLSA provenance
INFO[0008] create a symbolic link                        aqua_version=2.9.0 command=aqua env=linux/amd64 new_version=v2.16.4 package_name=aquaproj/aqua package_version=v2.16.4 program=aqua
aqua version 2.16.4 (fd3d2911afd67834ecdcf4c282fa294727cee9eb)
```

Tried to run aqua but doesn't work:

```sh
$ aqua g
aqua: command not found
``` 

On [another document page](https://aquaproj.github.io/docs/install), step 1 suggests to _Install the binary aqua in PATH_ but does not say how to do so. Where do I get the aqua binary? The installation output did not print this information. How is a binary "installed" in a PATH?

Another related questions: where can I find the instructions on how to uninstall aqua?

### Background of the question

I had this question while trying to install aqua for the first time. I would like to install aqua.

### Example Code

```console
$ 
```

Configuration

```yaml

```

### Reference

*

**Comment by suzuki-shunsuke** [comment, marked as answer]
Please set the environment variable `PATH` like this.

```sh
export PATH="${AQUA_ROOT_DIR:-${XDG_DATA_HOME:-$HOME/.local/share}/aquaproj-aqua}/bin:$PATH"
```

aqua-installer installs aqua to `${AQUA_ROOT_DIR:-${XDG_DATA_HOME:-$HOME/.local/share}/aquaproj-aqua}/bin/aqua`.

**Reply by lackovic** [reply]
I did that and got aqua to print the `root-dir`:

```sh
$ export PATH="${AQUA_ROOT_DIR:-${XDG_DATA_HOME:-$HOME/.local/share}/aquaproj-aqua}/bin:$PATH"
$ aqua root-dir
/home/marco/.local/share/aquaproj-aqua
```

but running `aqua g` produces the following error:

```sh
$ aqua g
FATA[0000] aqua failed    aqua_version=2.16.4 env=linux/amd64 error="configuration file isn't found" program=aqua
```

Again, regarding my related question: where can I find the instructions on how to uninstall aqua?

**Reply by suzuki-shunsuke** [reply]
> error="configuration file isn't found"

You need to create configuration files.

```sh
aqua init
```

About the location of configuration files, please see the following documents.

- https://aquaproj.github.io/docs/tutorial/config-path
- https://aquaproj.github.io/docs/reference/config/#configuration-file-path
- https://aquaproj.github.io/docs/tutorial/global-config

How about learning aqua with the tutorial?

https://aquaproj.github.io/docs/tutorial/

> where can I find the instructions on how to uninstall aqua?

https://aquaproj.github.io/docs/reference/uninstall/

**Comment by suzuki-shunsuke** [comment]
aqua-installer v2.2.0 is out 🎉

https://github.com/aquaproj/aqua-installer/releases/tag/v2.2.0

`aqua-installer` v2.2.0 outputs the following guide.

```
===============================================================
[INFO] aqua is installed into /root/.local/share/aquaproj-aqua/bin/aqua
[INFO] Please add the path to the environment variable "PATH"
[INFO] export PATH=${AQUA_ROOT_DIR:-${XDG_DATA_HOME:-$HOME/.local/share}/aquaproj-aqua}/bin:$PATH
===============================================================
```
